File: src/main/java/com/nerdearla/workshop/qr/validator/QRValidator.java
package com.nerdearla.workshop.qr.validator;

import com.nerdearla.workshop.qr.QR;
import com.nerdearla.workshop.qr.error.DisabledQrError;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

@Component
public class QRValidator {

    private static final Logger LOGGER = LoggerFactory.getLogger(QRValidator.class);

    public void validate(QR qr) {
        if(!qr.getEnabled()) {
            LOGGER.error("qr {} disabled", qr.getId());
            throw new DisabledQrError();
        }

        LOGGER.info("qr {} is enabled", qr.getId());
    }
}

File: src/main/kotlin/com/nerdearla/workshop/qr/validator/QrValidator.kt
package com.nerdearla.workshop.qr.validator

import com.nerdearla.workshop.qr.QR
import com.nerdearla.workshop.qr.error.DisabledQrError
import com.nerdearla.workshop.shared.utils.CompanionLogger
import org.springframework.stereotype.Component

@Component
class QrValidator {

    fun validate(qr: QR) {
        validateIsEnabled(qr)
    }

    private fun validateIsEnabled(qr: QR) {
        if (qr.enabled.not()) {
            log.error("qr {} is disabled", qr.id)
            throw DisabledQrError()
        }
        log.info("qr {} is enabled", qr.id)
    }

    companion object : CompanionLogger()
}
